lib/build.js

```diff
@@ -1,208 +1,133 @@
-                    }
-                }])
-            )
-            .then(() => nodeify.success(hoek.applyToDefaults({ id }, initialBuildData), callback))
-            .catch((err) => nodeify.fail(err, callback));
+        return job;
     }
 
     /**
-     * Gets all the builds for a jobId
-     * @method getBuildsForJobId
-     * @param  {Object}   config                Config object
-     * @param  {String}   config.jobId          The jobId of the build to filter for
-     * @param  {Function} [callback]            The callback object to return the stream to
-     * @return {Promise}                        If no callback is given, a Promise is returned
+     * Lazy load a user model
+     * @property user
+     * @return {Promise}    Resolves to the user associated with this build
      */
-    getBuildsForJobId(config, callback) {
-        const listConfig = {
-            params: {
-                jobId: config.jobId
-            },
-            paginate: {
-                count: 25, // This limit is set by the matrix restriction
-                page: 1
-            }
-        };
+    get user() {
+        delete this.user;
+        const factory = new UserFactory(this.datastore);
+        const user = factory.get({ username: this.username });
+
+        // ES6 has weird getters and setters in classes,
+        // so we redefine the pipeline property here to resolve to the
+        // resulting promise and not try to recreate the factory, etc.
+        Object.defineProperty(this, 'user', {
+            enumerable: true,
+            value: user
+        });
+
+        return user;
+    }
 
-        return this.list(listConfig)
-            .then((records) => {
-                records.sort((build1, build2) => build1.number - build2.number);
+    /**
+     * Start this build
+     * @method start
+     * @return {Promise}
+     */
+    start() {
+        const userFactory = new UserFactory(this.datastore);
 
-                return nodeify.success(records, callback);
-            })
-            .catch((err) => nodeify.fail(err, callback));
+        return this.job.then(job =>
+            job.pipeline.then(pipeline =>
+                // Start the build
+                nodeify.withContext(this.executor, 'start', [{
+                    buildId: this.id,
+                    container: this.container,
+                    jobId: job.id,
+                    jobName: job.name,
+                    pipelineId: pipeline.id,
+                    scmUrl: pipeline.scmUrl
+                }])
+                .then(() =>
+                    // fetch users
+                    Promise.all([
+                        this.user,
+                        userFactory.get({ username: pipeline.getAdmin() })
+                    ]).then(users => {
+                        const buildUser = users[0];
+                        const adminUser = users[1];
+
+                        return nodeify.withContext(githubHelper, 'run', [{
```

Comment:
How come we need to wrap with nodeify here for githubHelper.run but not other places?